lib/dai-plugin-mcd/test/SavingsService.spec.js

```diff
@@ -0,0 +1,94 @@
+import { mcdMaker, setupCollateral } from './helpers';
+import { ServiceRoles } from '../src/constants';
+import BigNumber from 'bignumber.js'
+import { MDAI, ETH } from '../src/index';
+import { stringToBytes } from '../src/utils';
+import { RAD } from '../src/constants'
+
+let service, maker, dai, proxyAddress;
+
+describe('Savings Service', () => {
+  async function makeSomeDai(amount) {
+    const cdpMgr = await maker.service(ServiceRoles.CDP_MANAGER);
+    await setupCollateral(maker, 'ETH-A', { price: 150, debtCeiling: 50 });
+    await cdpMgr.openLockAndDraw('ETH-A', ETH(1), MDAI(amount));
+  }
+
+  beforeAll(async () => {
+    maker = await mcdMaker();
+    service = maker.service(ServiceRoles.SAVINGS);
+    dai = maker.getToken(MDAI);
+
+    proxyAddress = await maker.service('proxy').ensureProxy();
+    await dai.approveUnlimited(proxyAddress)
+  });
+
+  afterAll(async () => {
+    await maker.service('allowance').removeAllowance('MDAI', proxyAddress);
+  })
+
+  beforeEach(async () => {
+    const amountRemaining = await service.balance();
+    if (amountRemaining.gt(0)) await service.exit(MDAI(amountRemaining));
+  })
+
+  test('get dai savings rate', async () => {
+    const dsr = await service.getDaiSavingsRate();
+
+    expect(maker.service('web3')._web3.utils.isBigNumber(dsr)).toBe(true);
```

Comment:
let's explicitly test for equality with the expected value--yes, it's more brittle against testchain config changes, but it catches order-of-magnitude errors.

Reply:
Good point.